tools/gen-intrinsics-specs.py

```diff
@@ -688,19 +758,32 @@ def recurse_print_to_rst(item, section_level_list, headers=__intrinsic_table_hea
     +=============+========================+=======================+==========+===========================+
     |           1 |                      2 |                     3 |        4 |                         5 |
     +-------------+------------------------+-----------------------+----------+---------------------------+
+    >>> print(recurse_print_to_rst(item, ['#','##'], tablefmt='pdf'))
+    <BLANKLINE>
+    # Section 1
+    <BLANKLINE>
+    ## Section 1.1
+    <BLANKLINE>
+    Text for Section 1.1
+    <BLANKLINE>
+    +-------------+------------------------+-----------------------+----------+---------------------------+
+    |   Intrinsic |   Argument preparation |   AArch64 Instruction |   Result |   Supported architectures |
+    +=============+========================+=======================+==========+===========================+
+    |           1 |                      2 |                     3 |        4 |                         5 |
+    +-------------+------------------------+-----------------------+----------+---------------------------+
     """
     key, value = item
     if is_intrinsic_table(item):
-        return "\n" + str(tbl.tabulate(value, headers=headers, tablefmt=tablefmt))
+        return "\n" + str(tbl.tabulate(value, headers=headers, tablefmt=(tablefmt if tablefmt != "pdf" else "grid")))
 
     body = ""
     if start_new_section(item):
         title, *rest = section_level_list
 
-        if tablefmt == "grid" or tablefmt == "rst":
+        if tablefmt == "rst" or tablefmt == "grid":
             body += f"\n{key}"
             body += "\n" + title * len(key) + ""
-        elif tablefmt == "github":
+        elif tablefmt == "github" or tablefmt == "pdf":
```

Comment:
This is too convoluted.

I would do one `if` statement for each of the `--source-syntax` passed at command line
```
if tablefmt == 'grid':
    ....
elif tablefmt == 'markdown'
    ....
elif tablefmt == 'pdf':
    .....
```